resources/secrets/src/lib.rs

```diff
@@ -2,25 +2,36 @@ use std::collections::BTreeMap;
 
 use async_trait::async_trait;
 
-use shuttle_service::{Error, Factory, ResourceBuilder};
+use serde::{Deserialize, Serialize};
+use shuttle_service::{Error, Factory, ResourceBuilder, Type};
 
+#[derive(Serialize)]
 pub struct Secrets;
 
 /// Get a store with all the secrets available to a deployment
 #[async_trait]
 impl ResourceBuilder<SecretStore> for Secrets {
+    const TYPE: Type = Type::Secrets;
+
+    type Output = SecretStore;
+
     fn new() -> Self {
         Self {}
     }
 
-    async fn build(self, factory: &mut dyn Factory) -> Result<SecretStore, Error> {
+    async fn build(build_data: &Self::Output) -> Result<SecretStore, crate::Error> {
+        Ok(build_data.clone())
+    }
+
+    async fn output(self, factory: &mut dyn Factory) -> Result<Self::Output, crate::Error> {
         let secrets = factory.get_secrets().await?;
```

Comment:
Maybe this can now read the secrets from the filesystem rather than getting them from factory? This means we can remove the secrets from the factory.

Reply:
But maybe we would still want to access historical secrets?

Reply:
I think storing them on the filesystem rather than the database makes sense, that way they reflect the latest build. But I'm not sure if that will be an issue when we allow setting secrets from the console, I feel like in that scenario having them in the DB would make more sense. :thinking:

Reply:
I'm also thinking maybe they should not be a mutable resource